com.unity.multiplayer.mlapi/Runtime/Messaging/RpcQueue/RpcQueueProcessor.cs

```diff
@@ -227,35 +224,42 @@ private void SendFrameQueueItem(RpcFrameQueueItem queueItem)
-                    PerformanceDataManager.Increment(ProfilerConstants.NumberOfRPCsSent);
-                    ProfilerStatManager.BytesSent.Record((int)queueItem.StreamSize);
-                    ProfilerStatManager.RpcsSent.Record();
-                    break;
-                }
-                case RpcQueueContainer.QueueItemType.ClientRpc:
-                {
-                    foreach (ulong clientid in queueItem.ClientNetworkIds)
                     {
-                        NetworkManager.Singleton.NetworkConfig.NetworkTransport.Send(clientid, queueItem.MessageData, queueItem.NetworkChannel);
+                        m_RpcQueueContainer.m_NetworkManager.NetworkConfig.NetworkTransport.Send(queueItem.NetworkId, queueItem.MessageData, queueItem.NetworkChannel);
 
                         //For each packet sent, we want to record how much data we have sent
                         PerformanceDataManager.Increment(ProfilerConstants.NumberBytesSent, (int)queueItem.StreamSize);
+                        PerformanceDataManager.Increment(ProfilerConstants.NumberOfRPCsSent);
                         ProfilerStatManager.BytesSent.Record((int)queueItem.StreamSize);
+                        ProfilerStatManager.RpcsSent.Record();
+                        break;
                     }
+                case RpcQueueContainer.QueueItemType.ClientRpc:
+                    {
+                        foreach (ulong clientid in queueItem.ClientNetworkIds)
+                        {
+                            m_RpcQueueContainer.m_NetworkManager.NetworkConfig.NetworkTransport.Send(clientid, queueItem.MessageData, queueItem.NetworkChannel);
+
+                            //For each packet sent, we want to record how much data we have sent
+                            PerformanceDataManager.Increment(ProfilerConstants.NumberBytesSent, (int)queueItem.StreamSize);
+                            ProfilerStatManager.BytesSent.Record((int)queueItem.StreamSize);
+                        }
 
-                    //For each client we send to, we want to record how many RPCs we have sent
-                    PerformanceDataManager.Increment(ProfilerConstants.NumberOfRPCsSent, queueItem.ClientNetworkIds.Length);
-                    ProfilerStatManager.RpcsSent.Record(queueItem.ClientNetworkIds.Length);
+                        //For each client we send to, we want to record how many RPCs we have sent
+                        PerformanceDataManager.Increment(ProfilerConstants.NumberOfRPCsSent, queueItem.ClientNetworkIds.Length);
+                        ProfilerStatManager.RpcsSent.Record(queueItem.ClientNetworkIds.Length);
 
-                    break;
-                }
+                        break;
+                    }
             }
         }
+
+        internal RpcQueueProcessor(RpcQueueContainer rpcQueueContainer)
+        {
+            m_RpcQueueContainer = rpcQueueContainer;
+        }
```

Comment:
should we move this up to the top? (styling)

Reply:
It is fine...